### PR TITLE
added highlighting

### PR DESCRIPTION
**Pull request by javasoze**
screen shot 2017-11-18 at 11 08 34 am

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -25,6 +25,8 @@
         'setuptools>=12.2',
         'simplejson==3.12.0',
         'terminaltables==3.1.0',
+        'colorama==0.3.7',
```
Comment (PureWhiteWu) [inline setup.py]: 1. Should add them in `setup.py` or users will raise `DistributionNotFound` error.
2. Please add these to requirements according to the *alphabetical order*
e.g. colorama should be at the top of file before configparser, termcolor should be between terminaltables and simplejson

File: tools/resultview.py [hunk 2]
```diff
@@ -183,6 +185,38 @@ def print_search_stats(result):
...
+        if highlightEntity:
+            return highlightEntity["fields"][field]
+        return None
+    except Error:
+        traceback.print_exc()
+        return None
+
+
+def get_hl_stored_entities(hit):
+    try:
```
Comment (PureWhiteWu) [inline tools/resultview.py]: why need try except here?
what exception may be raised?

File: tools/resultview.py [hunk 3]
```diff
@@ -183,6 +185,38 @@ def print_search_stats(result):
                                                             result["request"]["useApproximation"]))
 
 
+def get_hl_field_entities(hit, field):
+    try:
```
Comment (PureWhiteWu) [inline tools/resultview.py]: same

File: tools/resultview.py [hunk 4]
```diff
@@ -183,6 +185,38 @@ def print_search_stats(result):
...
+        highlightEntity = None
+        for entity in entities:
+            if entity["entityType"] == "highlight":
+                highlightEntity = entity
+                break
+        if highlightEntity:
+            return highlightEntity["fields"][field]
+        return None
+    except Error:
+        traceback.print_exc()
```
Comment (PureWhiteWu) [inline tools/resultview.py]: print the traceback after you catch the exception is meaningless

File: utils/highlight.py [hunk 5]
```diff
@@ -0,0 +1,30 @@
+from colorama import init
+from termcolor import colored
+
+init()
+
+
+def highligh_str(s, indicesList):
+    if not s or not indicesList or len(indicesList) == 0:
+        return s
+    try:
```
Comment (PureWhiteWu) [inline utils/highlight.py]: same

File: tools/resultview.py (hunk 2 above)
Comment (javasoze) [reply tools/resultview.py]: json may not have the elements

File: tools/resultview.py (hunk 4 above)
Comment (javasoze) [reply tools/resultview.py]: you suggest just call pass?

File: utils/highlight.py [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+        temps = None
+        i = 0
+        for indices in indicesList:
+            start = indices["offset"]
+            length = indices["length"]
+            if length > 0:
+                if temps:
+                    t = temps + s[i:start]
```
Comment (PureWhiteWu) [inline utils/highlight.py]: why not `temps = temps + s[i : start]`?

File: utils/highlight.py [hunk 7]
```diff
@@ -0,0 +1,30 @@
+from colorama import init
+from termcolor import colored
+
+init()
+
+
+def highligh_str(s, indicesList):
+    if not s or not indicesList or len(indicesList) == 0:
```
Comment (PureWhiteWu) [inline utils/highlight.py]: if `len(indicesList) == 0` , `not list` equals `True`

File: tools/resultview.py (hunk 2 above)
Comment (PureWhiteWu) [reply tools/resultview.py]: then should use `if KEY in xxx` or `if KEY not in xxx`......

File: tools/resultview.py [hunk 8]
```diff
@@ -183,6 +185,38 @@ def print_search_stats(result):
...
+            return highlightEntity["fields"][field]
+        return None
+    except ValueError:
+        traceback.print_exc()
+        return None
+
+
+def get_hl_stored_entities(hit):
+    try:
+        entities = hit["payload"]["entities"]
```
Comment (PureWhiteWu) [inline tools/resultview.py]: repeat code for 190-195, can split out to be a function.

File: tools/resultview.py [hunk 9]
```diff
@@ -215,7 +249,13 @@ def print_hits(fieldnames, hits, hide_border_and_title=False, show_raw=True):
                     if field in hit_fields.keys():
                         val = hit_fields[field]
                     if val is not None and len(val) > 0:
-                        data_array.append(val[0])
+                        hl = get_hl_field_entities(hit, field)
+                        fieldVal = None
```
Comment (PureWhiteWu) [inline tools/resultview.py]: `fieldVal = highligh_str(val[0], hl[0]) if h1 else val[0]` is more pythonic

**Comment by PureWhiteWu** [comment]
fixed #77

**Comment by javasoze** [comment]
LGTM, will merge.

**Pull request merged**